api/src/main/scala/org/ensime/api/incoming.scala

```diff
@@ -53,7 +53,7 @@ final case class TypecheckFileReq(fileInfo: SourceFileInfo) extends RpcAnalyserR
 /**
  * Response with a `VoidResponse`.
  */
-final case class UnloadModuleReq(module: String) extends RpcAnalyserRequest
+final case class UnloadModuleReq(moduleId: EnsimeProjectId) extends RpcAnalyserRequest
```

Comment:
If we added a custom marshaller we would have the configuration default to `"compile"`.
There could be cases when user might want to typecheck files in the `"test"` configuration as well and might fail later. (say) during resolution related to `ScalaTest` which wasn't among the `libraryJars` in the `"compile"` configuration. Currently since `compileDeps` and `testDeps` are in the same `EnsimeModule` identifiable through `moduleName`it wasn't be a problem. Right?